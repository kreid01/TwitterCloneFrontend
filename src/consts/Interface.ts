export type IPost = {
    id:number,
    userName:string
    userAt:string
    userImg:string,
    postTextBody:string
    postMedia:string
    postDate:string
    commentCount:number
    likeCount:number
    retweetCount:number
    isLiked?:boolean
    isRetweeted?:boolean
    comments?: Array<Comment>
}

export type IComment = {
    userAt: string
    userName: string
    userImg: string 
    commentBody: string
    commentMedia?: string
    commentDate: string
}

export type INewComment = {
        commentBody: string;
        commentMedia: string;
        userAt: string;
        userName: string;
        userImg: string;
}