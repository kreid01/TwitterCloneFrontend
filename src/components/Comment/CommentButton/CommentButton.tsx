import React from "react";
import { IPost, INewComment } from "../../../consts/Interface";

interface Props {
  handleReply: (Post: IPost, newComment: INewComment) => void;
  post: IPost;
  newComment: INewComment;
  label: string;
}

export const CommentButton: React.FC<Props> = ({
  handleReply,
  post,
  newComment,
  label,
}) => {
  return (
    <button
      data-testid="replyButton"
      onClick={() => handleReply(post, newComment)}
      className="button-primary mb-3"
    >
      {label}
    </button>
  );
};
