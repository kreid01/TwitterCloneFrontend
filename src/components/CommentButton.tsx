import React from 'react'
import {IPost, INewComment} from '../consts/Interface'

interface Props  {
    handleReply: (Post: IPost, newComment: INewComment ) => void
    post: IPost
    newComment: INewComment
}

export const CommentButton: React.FC<Props> = ({handleReply, post, newComment}) => {

    return (
        <button onClick={() => handleReply(post, newComment)} className='button-primary mb-3'>Reply</button>
    )
} 