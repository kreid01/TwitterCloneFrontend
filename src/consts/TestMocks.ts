export const testPost = {
  id: 1,
  userName: "test-name",
  userAt: "testAt",
  userImg: "testImg",
  postTextBody: "testBody",
  postMedia: "test-media",
  postDate: "testDate",
  commentCount: 1,
  likeCount: 1,
  retweetCount: 1,
  isRetweeted: false,
  isLiked: false,
  retweetedBy: [1],
  comments: [],
};

export const testComment = {
  id: 1,
  userName: "test-name",
  userAt: "testAt",
  userImg: "testImg",
  commentBody: "testBody",
  commentMedia: "test-media",
  commentDate: "test-date",
};
export const testPosts = [testPost, testPost];

export const testUser = {
  userId: 1,
  userName: "test-name",
  userAt: "testAt",
  userImg: "testImg",
  userCoverImg: "testCoverImg",
  userEmail: "testEmail",
  postsIds: [1, 2],
  joinDate: "testDate",
  following: [1, 2],
  followers: [1, 2],
  email: "",
  password: "",
};
