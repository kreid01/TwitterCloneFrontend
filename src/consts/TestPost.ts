export const testPost = {
    id:1,
    userName:'testName',
    userAt:'testAt',
    userImg:'testImg',
    postTextBody:'testBody',
    postMedia:'testMedia',
    postDate:'testDate',
    commentCount:1,
    likeCount:1,
    retweetCount:1
}