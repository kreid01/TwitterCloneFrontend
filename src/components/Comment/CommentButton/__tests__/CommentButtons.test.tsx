/* eslint-disable testing-library/prefer-screen-queries */
import React from 'react'
import { CommentButton } from '../CommentButton'
import { testPost, testComment } from 'consts/TestMocks';

import { screen, render, cleanup, fireEvent } from '@testing-library/react';
import "@testing-library/jest-dom";

import renderer from 'react-test-renderer'

const mockFn = jest.fn();

const handleTweet = jest.fn()

afterEach(cleanup)
it("render without crashing", () => {
    render(<CommentButton
    label='Reply'
    newComment={testComment}
    post={testPost}
    handleReply={mockFn}/>)
})

it("renders button correctly", () => {
    render(<CommentButton
        label='Reply'
        newComment={testComment}
        post={testPost}
        handleReply={mockFn}/>)
    const element =  screen.getByTestId("replyButton")
    expect(element).toHaveTextContent('Reply')
})

it("calls handleTweeet function when clicked", () => {
    render(<CommentButton
        label='Reply'
        newComment={testComment}
        post={testPost}
        handleReply={mockFn}/>)
    fireEvent.click(screen.getByTestId("replyButton"))
    expect(mockFn).toHaveBeenCalled()
})
 
it("matched snapshot", () => {
    const tree = renderer.create(
    <CommentButton
        label='Reply'
        newComment={testComment}
        post={testPost}
        handleReply={mockFn}/>).toJSON()
    expect(tree).toMatchSnapshot()
})
