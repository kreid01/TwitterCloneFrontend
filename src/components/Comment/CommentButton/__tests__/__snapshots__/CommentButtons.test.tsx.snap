// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matched snapshot 1`] = `
<button
  className="button-primary mb-3"
  data-testid="replyButton"
  onClick={[Function]}
>
  Reply
</button>
`;
